File: src/compiler/parser/visitors/OperatorTypedVisitor.java
package compiler.parser.visitors;

import static compiler.lexer.token.OperatorToken.*;

public interface OperatorTypedVisitor<T> {
  T visit(LessThan token);

  T visit(GreaterThan token);

  T visit(Asterisk token);

  T visit(Equal token);

  T visit(Minus token);

  T visit(Plus token);

  T visit(Ampersand token);

  T visit(Arrow token);

  T visit(EqualEqual token);

  T visit(NotEqual token);

  T visit(LessThanOrEqual token);

  T visit(GreaterThanOrEqual token);

  T visit(BitShiftLeft token);

  T visit(BitShiftRight token);
}

File: src/compiler/parser/visitors/OperatorTypedVisitor.kt
package compiler.parser.visitors

import compiler.lexer.token.OperatorToken.*

interface OperatorTypedVisitor<T> {
    fun visit(token: LessThan): T
    fun visit(token: GreaterThan): T
    fun visit(token: Asterisk): T
    fun visit(token: Equal): T
    fun visit(token: Minus): T
    fun visit(token: Plus): T
    fun visit(token: Ampersand): T
    fun visit(token: Arrow): T
    fun visit(token: EqualEqual): T
    fun visit(token: NotEqual): T
    fun visit(token: LessThanOrEqual): T
    fun visit(token: GreaterThanOrEqual): T
    fun visit(token: BitShiftLeft): T
    fun visit(token: BitShiftRight): T
    fun visit(token: Caret): T
    fun visit(token: ForwardSlash): T
}
